### ISSUE TITLE
Failed to execute docker container built from the Dockerfile due to systemd quirks

### ISSUE THREAD

**Issue by fooryo** [ffplayout/ffplayout#578]
**Describe the bug**

After building the docker image with `docker build -t ffplayout-image .` and run it with `docker run -ti --name ffplayout -v /sys/fs/cgroup:/sys/fs/cgroup:ro --cap-add SYS_ADMIN -p 8787:8787 ffplayout-image` as the README.md instructs, it returns:
```
systemd 252-18.el9 running in system mode (+PAM +AUDIT +SELINUX -APPARMOR +IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS -FIDO2 +IDN2 -IDN -IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY +P11KIT -QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD -BPF_FRAMEWORK +XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified)
Detected virtualization docker.
Detected architecture x86-64.

Welcome to AlmaLinux 9.3 (Shamrock Pampas Cat)!

Initializing machine ID from random generator.
Failed to create /init.scope control group: Read-only file system
Failed to allocate manager object: Read-only file system
[!!!!!!] Failed to allocate manager object.
Exiting PID 1...

```

**To Reproduce**

1. `docker build -t ffplayout-image .`
2. `docker run -ti --name ffplayout -v /sys/fs/cgroup:/sys/fs/cgroup:ro --cap-add SYS_ADMIN -p 8787:8787 ffplayout-image`

**Expected behavior**

well, i dont know since it is my first time.

**Desktop/Server/Software (please complete the following information):**

 - OS: Manjaro Linux 6.7.7-1-MANJARO KDE Plasma 5.27.11
- ffplayout version: docker one
- ffmpeg version: docker one
- are you using the current master of ffplayout? yes

**Config Settings:**

- command line arguments: docker one
- config fie: docker one

**Comment by lukeomatik** [comment]
> I would like to share this resource:
> 
> https://serverfault.com/questions/1053187/systemd-fails-to-run-in-a-docker-container-when-using-cgroupv2-cgroupns-priva/1054414#1054414
> 
> but since I dont really know systemd/slices/cgroups I wont recommend to enable switches

should it be a workaround? I didn't manage to make it runs

**Comment by jb-alvarado** [comment]
You can use this solution: https://serverfault.com/a/1087467/387878

It is enough to replace the `VOLUME` statement with: `VOLUME [ "/tmp", "/run", "/run/lock" ]`

The you can run the container with:

```BASH
docker run -ti --name ffplayout --privileged --security-opt seccomp=unconfined --cgroup-parent=docker.slice --cgroupns private --tmpfs /tmp --tmpfs /run --tmpfs /run/lock --cap-add SYS_ADMIN -p 8787:8787 ffplayout-image
```

This needs to be testet more, maybe not all parameters a needed and we have to see that this works also on other systems like ubuntu/debian, fedora and rocky. If it runs on all platforms we can update the docs with the new settings.

**Comment by fooryo** [comment]
@jb-alvarado I wanna be honest with you just because I really liked ffplayout:

I don't think I will use this docker image anymore. I don't really see nor understand why a docker container should use systemd inside of it nor interact with the host's systemd / environment. I can't see the reason and I know maybe it's my fault.

Looking forward to it in the future!

**Comment by jb-alvarado** [comment]
@fooryo I totally agree with you and we don't run ffplayout in a container, but some people like to have this options and I got a pull request for that. The other reason is, that there is a image from Nvidia that can be use for HW encoding.

The use of systemd in the container is not that optimal, but that is the simplest option to be able to control ffplayout with ffpapi. I know it goes a bit against the container philosophy to run two process in one container, but controlling a process in container A from another process in container B is not that easy without process management.

I have update now the Definitions and Readme.

**Comment by fooryo** [comment]
@jb-alvarado I know nothing about nvidia. For what is worth: I know that encoding a video inside a docker container using Intel QSV is done by passing the device using the `--device` flag.  And I guess something similar and straightforward is present for nvidia ( I read something for it when I was looking at Jellyfin on docker plus nvidia encoder )

About ffplayout + ffpapi: I honestly should re-read about your architecture and what ffpapi is able to do before asking this question but here it is: isnt there another way? like a REST(http)+websocket(json) server(listener) at the engine and ffpapi that connects to it and maybe serves also the Vue application ? could it be viable? Just asking and giving ideas, no more. I know basically nothing about proper IPC.

**Comment by jb-alvarado** [comment]
Personally I don't use HW encoding with NVIDIA device, but if you use it the easiest is with docket.

Currently ffpapi serves the frontend and I use simple IPC for controlling some things on the engine, but not start/stop/restart.

I have plans to switch to Tokio and control everything over a REST API, but then I have to rewrite all functions to be asynchronous. And this takes time. In the past this was not even possible because of a limitation in Rust, but I think they have fix that now.
